mautic/mautic
# Empty and Not Empty report date filters are not compatible with MySQL8

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

[//]: # ( Required: )
#### Description:
When a user uses an empty or not empty filter within a report after being upgraded to MySQL 8 the report will crash and not return results. This PR fixes no longer valid `date`/`datetime` comparison with an empty string in MySQL 8.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a new report with the following filters on the _Data_ tab:
    -  A field of _Data Type_ **Date** with a condition **empty**.
    -  A field of _Data Type_ **Date** with a condition **not empty**.
    -  A field of _Data Type_ **Date/Time** with a condition **empty**.
    -  A field of _Data Type_ **Date/Time** with a condition **not empty**.
    -  A field of _Data Type_ **Time** with a condition **empty**.
    -  A field of _Data Type_ **Time** with a condition **not empty**.
2. When you go to the report's view page, you get _500 Internal Server Error_ message and in the logs there is `General error: 1525 Incorrect DATE value` error.

#### Steps to test this PR:
1. Repeat the first step above.
2. When you go to the report's view page, there is no error.
3. Validate that empty/not empty filters work as expected for both date and non-date fields.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10194"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
